From b389dad03f43cdd62fb5bede6def4310a1ee0656 Mon Sep 17 00:00:00 2001
From: Sherry Sun <sherry.sun@nxp.com>
Date: Wed, 13 Jul 2022 13:56:54 +0800
Subject: [PATCH 228/274] LF-6593 arm64: dts: imx93-11x11-evk-lpuart.dts: add a
 new dts file for uart test

Add a new dts file for uart test on imx93-11x11-evk board.
We multiplex the lpuart5 and lpuart6 test ports from the GPIO pins of
J1001 interface on imx93-11x11-evk board.

Signed-off-by: Sherry Sun <sherry.sun@nxp.com>
Reviewed-by: Li Jun <jun.li@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  3 +-
 .../dts/freescale/imx93-11x11-evk-lpuart.dts  | 37 +++++++++++++++++++
 2 files changed, 39 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx93-11x11-evk-lpuart.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index e99e1e9eb..ac3776484 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -247,6 +247,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-evk.dtb imx93-11x11-evk-rpmsg.dtb \
 			  imx93-11x11-evk-i3c.dtb imx93-11x11-evk-rm67199.dtb \
 			  imx93-11x11-evk-inmate.dtb imx93-11x11-evk-root.dtb \
 			  imx93-11x11-evk-boe-wxga-lvds-panel.dtb imx93-11x11-evk-mqs.dtb \
-			  imx93-11x11-evk-aud-hat.dtb imx93-11x11-evk-flexio-i2c.dtb
+			  imx93-11x11-evk-aud-hat.dtb imx93-11x11-evk-flexio-i2c.dtb \
+			  imx93-11x11-evk-lpuart.dtb
 dtb-$(CONFIG_ARCH_S32) += s32v234-evb.dtb \
 			  s32v234-sbc.dtb
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-evk-lpuart.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-evk-lpuart.dts
new file mode 100644
index 000000000..17759dbe4
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-evk-lpuart.dts
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2022 NXP
+ */
+
+#include "imx93-11x11-evk.dts"
+
+&lpuart6 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart6>;
+	status = "okay";
+};
+
+/*
+ * uart test port1. Note: don't use Bluetooth at the sametime.
+ * J1001: Pin27--TX Pin28--RX Pin3--CTS Pin5--RTS
+ */
+&pinctrl_uart5 {
+	fsl,pins = <
+		MX93_PAD_GPIO_IO00__LPUART5_TX      0x31e
+		MX93_PAD_GPIO_IO01__LPUART5_RX      0x31e
+		MX93_PAD_GPIO_IO02__LPUART5_CTS_B   0x31e
+		MX93_PAD_GPIO_IO03__LPUART5_RTS_B   0x31e
+	>;
+};
+
+&iomuxc {
+/* uart test port2: J1001: Pin7--TX Pin29--RX Pin31--CTS Pin26--RTS */
+	pinctrl_uart6: uart6grp {
+		fsl,pins = <
+			MX93_PAD_GPIO_IO04__LPUART6_TX		0x31e
+			MX93_PAD_GPIO_IO05__LPUART6_RX		0x31e
+			MX93_PAD_GPIO_IO06__LPUART6_CTS_B	0x31e
+			MX93_PAD_GPIO_IO07__LPUART6_RTS_B	0x31e
+		>;
+	};
+};
-- 
2.25.1

